netbox/ipam/tests/test_ordering.py

```diff
@@ -0,0 +1,182 @@
+from django.test import TestCase
+
+from ipam.choices import IPAddressStatusChoices, PrefixStatusChoices
+from ipam.models import IPAddress, Prefix, VRF
+
+import netaddr
+
+
+class OrderingTestBase(TestCase):
+    vrfs = None
+
+    """
+    Setup the VRFs for the class as a whole
+    """
+    def setUp(self):
+        self.vrfs = (VRF(name="VRF A"), VRF(name="VRF B"), VRF(name="VRF C"))
+        VRF.objects.bulk_create(self.vrfs)
+
+    """
+    Perform the comparison of the queryset object and the object used to instantiate the queryset.
+    """
+    def _compare(self, queryset, objectset):
+        for i, obj in enumerate(queryset):
+            if isinstance(obj, Prefix):
+                self.assertEqual((obj.vrf, obj.prefix), (objectset[i]['vrf'], objectset[i]['prefix']))
+            elif isinstance(obj, IPAddress):
+                self.assertEqual((obj.vrf, obj.address), (objectset[i]['vrf'], objectset[i]['address']))
+
+
+class PrefixOrderingTestCase(OrderingTestBase):
+
+    """
+    This is for comparing the complex ordering test case
+    """
+    def _compare_complex(self, queryset, prefixes):
+        qsprefixes, regprefixes = [], []
+        for i, obj in enumerate(queryset):
+            qsprefixes.append(obj.prefix)
+        for pfx in prefixes:
+            regprefixes.append(pfx['prefix'])
+        self.assertEquals(qsprefixes, regprefixes)
+
+    """
+    This is a very basic test, which tests both prefixes without VRFs and prefixes with VRFs
+    """
+    def test_prefix_vrf_ordering(self):
+        # Setup VRFs
+        vrfa, vrfb, vrfc = self.vrfs
+
+        # Setup Prefixes
+        prefixes = (
+            {"status": PrefixStatusChoices.STATUS_CONTAINER, "vrf": None, "family": 4, "prefix": netaddr.IPNetwork('192.168.0.0/16')},
```

Comment:
Instead of defining all the attrs in dicts first, you can create `Prefix` instances directly in a list, and then pass that list to `Prefix.objects.bulk_create()`. This cuts down on boilerplate somewhat.

```python
prefixes = (
    Prefix(status=PrefixStatusChoices.STATUS_CONTAINER, vrf=None, family=4, prefix=netaddr.IPNetwork('192.168.0.0/16'),
    ...
)
Prefix.objects.bulk_create(prefixes).
```

Plus, you can reference the prefixes by their index on the `prefixes` list even after creation in the database if needed:

```python
prefixes[4].pk
```